src/drawer.canvasentry.js

```diff
@@ -159,26 +163,41 @@ export default class CanvasEntry {
     }
 
     /**
-     * Utility function to handle wave color arguments
+     * Utility function to determine the type and handle the argument passed to
+     * a color setting method
      *
-     * When the color argument type is a string, it will be returned as is.
-     * Otherwise, it will be treated as an array, and a canvas gradient will
-     * be returned
+     * Return type is dependent on the color argument's type
      *
      * @since 5.3.0
      * @param {CanvasRenderingContext2D} ctx Rendering context of target canvas
-     * @param {string|string[]} color Fill color for the wave canvas, or an array of colors to apply as a gradient
-     * @returns {string|CanvasGradient} Returns a string fillstyle value, or a canvas gradient
+     * @param {string|string[]|CanvasImageSource} color A CSS color value as a
+     * string, an array of CSS color value strings, or a Canvas Image Source
+     * (HTML Image, SVG Image, HTML Video, HTML Canvas, Image Bitmap or
+     * Offscreen Canvas element)
```

Comment:
i'm not sure if this functionality should be part of setWaveColor because its not a color but an image that is used (so sounds more like a texture).

Reply:
Agreed that passing an image to an option called "color" doesn't feel right.  Do you think there should be a new parameter on the Wavesurfer.create()?, such as waveImage & progressImage?

Reply:
I was working on creating options for the wave and progress waveforms that didn't end up setting an image style based on a value from an option property named "color".  I have the following changes for the parameters:

@property {string}  waveColor='#999'  The fill color of the waveform after the cursor.
@property {WaveformFillStyle} waveFillStyle={} Fill style of the waveform.`
 
@typedef {Object} WaveformFillStyle
@desc styles applied to the unplayed waveform
@property {string} style=can be color, gradient or image
@property {ImageFillStyle} image={} object describing image properties of the image type of waveform fill style
@property {string[]} gradient=[] array of color stops
@property {string} color=color value
@since 6.0.0

@typedef {Object} ImageFillStyle
@desc styles applied to the waveform when the style type is image
@property {CanvasImageSource} src=HTML Image, SVG Image, HTML
Video, HTML Canvas, Image Bitmap or Offscreen Canvas element used to fill the unplayed waveform after the cursor.
@property {string} repetition=repeat, repeat-x, repeat-y or no-repeat
@since 6.0.0

 so that WaveSurfer color, gradient and image styles can be declared as follows (which also includes msuhov's change request that the repeat parameter on the canvas image pattern object can be controlled) -

```
waveColor: '#555',
waveFillStyle: {
            style: 'image',
            image: {
                src: '',
                repetition: 'no-repeat'
            },
            gradient: ['blue', 'red'],
            color: '#999'
        },
```

and then allowing each to fallback to the next if something doesn't work out - ie if they provide all 3 - an image, gradient, and color, but something is wrong with the image, the gradient has an error in the color code, then it can fallback to the color.

Per your comment, I was looking to define something along the lines of setWaveImage/getWaveImage, but I can't find anywhere that actually calls getWaveColor/setWaveColor during instantiation.  Where should the logic that will call either getWaveColor, or getWaveImage, etc.. go?  As it needs access to the setup parameters.

Is this a path I should continue on?  Thanks!

Reply:
what about a new setWaveStyle and setProgressStyle? the old setWaveColor and setProgressColor would also use these new methods internally. So you could use either setWaveColor or setWaveStyle to change the color. Perhaps eventually these older methods can be deprecated but I don't mind having them around.

As to your suggested configuration settings; not sure yet, can you first add the new set[Wave/Progress]Style methods and we'll go from there.

Reply:
done